dali/python/nvidia/dali/ops.py

```diff
@@ -601,29 +601,103 @@ def _choose_device(inputs):
         return "gpu"
     return "cpu"
 
-def _target_uniform_dev(inputs, dev):
-    return inputs
-
-
-def _arithm_op(name, *inputs):
-        input_desc = ""
-        for i, input in enumerate(inputs):
-            input_desc += "&" + str(i)
-            if i < len(inputs) - 1:
-                input_desc += " "
-        expression_desc = "{}({})".format(name, input_desc)
-        dev = _choose_device(inputs)
-        # Create "instance" of operator
-        op = ArithmeticGenericOp(device = _choose_device(inputs), expression_desc = expression_desc)
-        # If we are on gpu, we must mark all inputs as gpu
-        if dev == "gpu":
-            dev_inputs = list(input.gpu() for input in inputs)
+def _is_integer_like(input):
+    if type(input) == int:
+        return True
+    if isinstance(input, Constant):
+        if input.dtype in [DALIDataType.INT8, DALIDataType.INT16, DALIDataType.INT32,
+                DALIDataType.INT64, DALIDataType.UINT8, DALIDataType.UINT16, DALIDataType.UINT32,
+                DALIDataType.UINT64]:
+            return True
+    return False
+
+def _is_real_like(input):
+    if type(input) == float:
+        return True
+    if isinstance(input, Constant):
+        if input.dtype in [DALIDataType.FLOAT16, DALIDataType.FLOAT, DALIDataType.FLOAT64]:
+            return True
+    return False
+
+def _to_type_desc(input):
+    if  type(input) == int:
+        return "int32"
+    if type(input) == float:
+        return "float32" # TODO(klecki): current DALI limitation
+    if isinstance(input, Constant):
+        dtype_to_desc = {
+            DALIDataType.INT8: "int8",
+            DALIDataType.INT16: "int16",
+            DALIDataType.INT32: "int32",
+            DALIDataType.INT64: "int64",
+            DALIDataType.UINT8: "uint8",
+            DALIDataType.UINT16: "uint16",
+            DALIDataType.UINT32: "uint32",
+            DALIDataType.UINT64: "uint64",
+            DALIDataType.FLOAT16: "float16",
+            DALIDataType.FLOAT: "float32",
+            DALIDataType.FLOAT64: "float64",
+        }
+        return dtype_to_desc[input.dtype]
+    raise TypeError(("Argument to arithmetic operation not supported. Got {}, expected "
+                "a return value from other DALI Operator or a constant value of type 'int', "
```

Comment:
`a return value from other DALI Operator` - but I don't see any check against it here?

Reply:
Copy-paste. Done.